metaswitch/common/mib.py

```diff
@@ -154,6 +160,42 @@ def table_test(stat):
 
         return self._table
 
+
+    def is_index_field (self):
+        """Determine if this is an index field or not by stepping back through
+        the ancestors inside the table.
+
+        Raises `LookupError` if the statistic is not in a table."""
+
+        field_name = self.get_info("SNMP NAME")
+
+        # Check that we're inside a table.
+        self.table()
+
+        for ancestor in self.ancestors():
+            logging.debug("Looking at Ancestor: %s" %
+                          ancestor.get_info("SNMP NAME"))
+            ancestor_index_string = ancestor.get_info("INDEX")
+            if ancestor_index_string:
+                # String should be of form "{ comma separated indices }" or
+                # blank if no indices. Check that there's an acutal string to
+                # parse then split it into separate elements and look for a
+                # match with the field name.
+                ancestor_index_string = ancestor_index_string[2:-2]
+                logger.debug("Checking if %s is in %s" %

```

Comment:
It's better to let the logger do the string substitution for you:
* It can avoid doing it if it doesn't need to.
* You don't have to worry about wrapping the args up in tuples.